src/stackdriver-nozzle/stackdriver/metrics_buffer.go

```diff
@@ -16,12 +16,83 @@
 
 package stackdriver
 
-import "reflect"
+import (
+	"context"
+	"github.com/mitchellh/hashstructure"
+	"reflect"
+	"sync"
+	"time"
+)
 
 type MetricsBuffer interface {
 	PostMetric(*Metric)
 }
 
+type timerMetricsBuffer struct {
```

Comment:
Yeah, so what are we actually doing? Seems like we're buffering metrics, culling them, and periodically flushing them. `autoCulledMetricsBuffer`?